### ISSUE TITLE
False positive: if_same_then_else 

### ISSUE THREAD

**Issue by alexheretic** [rust-lang-nursery/rust-clippy#2426]
I thought I'd submit this code as evidence of `if_same_then_else` not being particularly useful. I'm not sure this rule makes sense when you have a couple of `else if`s in there.

```rust
pub(super) fn combine_or_destroy(self, other: Self) -> Option<Self> {
    if other == self {
        // destroy
        None
    }
    else if other.from == self.from {
        // partial destroy
        Selection::between(other.to, self.to)
    }
    else if other.to == self.to {
        // partial destroy
        Selection::between(self.from, other.from)
    }
    else if other.from == self.to {
        // merge
        Selection::between(self.from, other.to)
    }
    else if other.to == self.from {
        // merge
        Selection::between(other.from, self.to)
    }
    else {
        None
    }
}
```
```
warning: this `if` has identical blocks
  --> compute/src/codeinput/selection.rs:69:14
   |
69 |           else {
   |  ______________^
70 | |             None
71 | |         }
   | |_________^
   |
   = note: #[warn(if_same_then_else)] on by default
note: same as this
  --> compute/src/codeinput/selection.rs:49:26
   |
49 |           if other == self {
   |  __________________________^
50 | |             // destroy
51 | |             None
52 | |         }
   | |_________^
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.184/index.html#if_same_then_else
```

**Comment by clarcharr** [comment]
Yeah, no, it doesn't.
